src/win/pipe.c

```diff
@@ -731,6 +831,20 @@ int uv_pipe_bind2(uv_pipe_t* handle,
     return UV_EINVAL;
   }
 
+  int use_uds_pipe = uv__should_use_uds_pipe(name);
```

Comment:
It will never be true, when `UV__ENABLE_WIN_UDS_PIPE` is not defined.

```c
static int uv__should_use_uds_pipe(const char *s) {
#if defined(UV__ENABLE_WIN_UDS_PIPE)
  /* Tell if the name is started by the named pipe prefix */
  if (strstr(s, pipe_prefix) == s) {
    return 0;
  }

  return 1;
#endif

  /* Disable uds on mingw */
  return 0;
}
```

Reply:
Yes, but what if the user passes a UDS path when not supported? we need to signal error.

Reply:
Oh, I was expecting the existing logic. If user pass a non-NamedPipe here and uds is not supported, it should already act like passed a bad pipe name in.

Because uds can be absolute and relative paths, when we support uds, we can only choose to believe that all previous 'bad-pipe' to be a possible valid uds pathname.

Do we need to explicit throw an error when uds is not supported? I think reusing the existing logic is suffice, just pretent we don't have such function?

Reply:
Let me be clear:
Previously on Windows:
`//./pipe/test` -> valid pipe
`bad-pipe` -> bad pipe

Now on Windows:
`//./pipe/test` -> valid pipe
`bad-pipe` -> valid uds

Now on Windows Mingw:
`//./pipe/test` -> valid pipe
`bad-pipe` -> ???

Reply:
IMO we probably should eliminate the concept of `bad-pipe` and consider all names without prefix to be a valid uds name. Thus, throwing an error will be reasonable if mingw not supported.

Reply:
> Now on Windows Mingw:
> `//./pipe/test` -> valid pipe
> `bad-pipe` -> ???

UV_ENOSYS

Reply:
SGTM, that also helps me rewrite the tests.